jerry-core/ecma/builtin-objects/ecma-builtin-regexp.c

```diff
@@ -84,54 +83,44 @@ ecma_builtin_regexp_dispatch_construct (const ecma_value_t *arguments_list_p, /*
   {
     if (ecma_is_value_undefined (flags_value))
     {
-      ret_value = ecma_copy_value (pattern_value);
+      return ecma_copy_value (pattern_value);
     }
-    else
-    {
-      ret_value = ecma_raise_type_error (ECMA_ERR_MSG ("Invalid argument of RegExp call."));
-    }
-  }
-  else
-  {
-    ecma_string_t *pattern_string_p = NULL;
-    ecma_string_t *flags_string_p = NULL;
-
-    ret_value = ecma_regexp_read_pattern_str_helper (pattern_value, &pattern_string_p);
 
-    if (ecma_is_value_empty (ret_value) && !ecma_is_value_undefined (flags_value))
-    {
-      ECMA_TRY_CATCH (flags_str_value,
-                      ecma_op_to_string (flags_value),
-                      ret_value);
+    return ecma_raise_type_error (ECMA_ERR_MSG ("Invalid argument of RegExp call."));
+  }
 
-      flags_string_p = ecma_get_string_from_value (flags_str_value);
-      ecma_ref_ecma_string (flags_string_p);
+  ecma_string_t *pattern_string_p = NULL;
+  ecma_value_t ret_value = ecma_regexp_read_pattern_str_helper (pattern_value, &pattern_string_p);
+  if (ECMA_IS_VALUE_ERROR (ret_value))
+  {
+    return ret_value;
+  }
```

Comment:
As the value of `ret_value` is overwritten later on, an assert on `ret_value` being undefined (or empty?) should be used here, too.